src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -316,8 +319,198 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+                    {
+                        dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                    }
+
+                    text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToUnixTimeSeconds().ToString(), text.Substring(match.Index + match.Value.Length));
+                }
+                catch (Exception)
+                {
+                    // If we can't parse the offset, just leave the variable in place
+                    // the error will be reported to the parser via "CanParse"
+                }
+            }
+        }
+        return text;
+    }
+
+    protected string ParseDateTime(string expression, MatchCollection matches, Regex pattern)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+
+        // We just pre-matched the prefix, so we need to re-match the full pattern to get the type, offset, and option
+        matches = pattern.Matches(text);
+
+        for (int i = matches.Count - 1; i >= 0; i--)
+        {
+            string format;
+            DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+            Match match = matches[i];
+            if (match.Groups.Count == 4)
+            {
+                try
+                {
+                    IFormatProvider formatProvider = Thread.CurrentThread.CurrentUICulture;
+
+                    Group type = match.Groups["type"];
+                    if (match.Groups["offset"].Success && match.Groups["option"].Success)
+                    {
+                        if (int.TryParse(match.Groups["offset"].Value, out int offset) && offset != 0)
+                        {
+                            dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                        }
+                    }
+
+                    if (string.Equals(type.Value, "rfc1123", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // For RFC1123, we want to be sure to use the invariant culture,
+                        // since we are potentially overriding the format for local date time
+                        // we should explicitly set the format provider to invariant culture
+                        formatProvider = CultureInfo.InvariantCulture;
+                        format = "r";
+                    }
+                    else if (string.Equals(type.Value, "iso8601", StringComparison.OrdinalIgnoreCase))
+                    {
+                        format = "o";
+                    }
+                    else
+                    {
+                        format = type.Value.Substring(1, type.Value.Length - 2);
+                    }
+
+                    text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToString(format, formatProvider), text.Substring(match.Index + match.Value.Length));
+                }
+                catch (Exception)
```

Comment:
It's preferable to catch the specific exception types that we think are possible here. Any other exceptions thrown here are a bug that the current approach would hide.